input/input_driver.c

```diff
@@ -1766,17 +1766,17 @@ int16_t input_joypad_analog(const input_device_driver_t *drv,
       {
          int16_t digital_left  = 0;
          int16_t digital_right = 0;
-         uint64_t key_minus    = bind_minus->joykey;
-         uint64_t key_plus     = bind_plus->joykey;
+         uint16_t key_minus    = bind_minus->joykey;
+         uint16_t key_plus     = bind_plus->joykey;
 
          if (key_minus == NO_BTN)
             key_minus = joypad_info.auto_binds[ident_minus].joykey;
          if (key_plus == NO_BTN)
             key_plus = joypad_info.auto_binds[ident_plus].joykey;
 
-         if (drv->button(joypad_info.joy_idx, (uint16_t)key_minus))
+         if (drv->button(joypad_info.joy_idx, key_minus))
```

Comment:
What is `drv->button`? Could it be beneficial (now, or for the future) to instead accept the full `uint64_t`?

Reply:
16 bits allows 65536 buttons, not 16, that callback asks "is button X pressed", RetroArch already only supports that many buttons in the input driver as it takes a uint16_t thus the cast.

I think those types where sized wrong to start with.
As for it being obsessive and nit picky, I agree with you, but anything less then perfect is still wrong, just because you have the extra memory doesnt mean you should waste it.
Also all out devices dont have gigs of memory, RetroArch runs on the PSP which has as low as 32mb.

Reply:
I agree. Given the new facts, this type was oversized.

*I still wouldn't argue with "wasting memory" (even for platforms like PSP), but simply using the wrong tool for the job (battering ram to hammer a nail).*